### PR TITLE
783 details summary buttons hover and redesign

### LINKED ISSUE #783: Details (summary) buttons hover and redesign
# Feature Request

## Description
Give the details buttons of the config hover styles. 
- Change the `details` and `summary` elements to custom buttons. Turns out semantic html isn't always better. Here's [an article](https://www.scottohara.me/blog/2022/09/12/details-summary.html) on this.
- Make the visually-hidden h2 "ScottBrew System Access" visible
- (if possible) add an "expand all/collapse all" button right next to the heading, allowing users to expand/collapse all content at once, to make the content searchable.

## Additional context
- Some screen readers read out the small triangles that come with detail and summary elements by default. You could remove them using css, but that then sometimes removes the state indication (expanded/collapsed) as well. 

## Acceptance criteria
**GIVEN** a screen reader user moves through the configurations
**WHEN** they land on the "Defence Configuration" or "details" button
**THEN** the visual icon is not announced, but the state of the button (expanded/collapsed) is announced .

### PR DESCRIPTION
**Pull request by chriswilty**
## Description

Reimplementation of the native expanding `<details>` element, to be more accessible and screen-reader friendly.

**TIP: hide whitespace when reviewing!**

There are a few things wrong with the original, including the arrow icons being read out by screen readers, plus the way the control acts as a button but contains its own descriptive header and content, which was making life difficult as we also have a control toggle to turn a defence on/off, which we want always to be visible even when the details are collapsed.

All the glory belongs to @dhinrichs-scottlogic, I just tidied up a few things here and there.

Resolves #783 

## Screenshots

image

image

## Checklist

Have you done the following?

- [x] Linked the relevant Issue
- [x] Added tests
- [x] Ensured the workflow steps are passing

### CODE REVIEW

File: frontend/src/components/ThemedInput/ThemedNumberInput.css [hunk 1]
```diff
@@ -7,5 +7,6 @@
 
 /* hide arrows in firefox */
```
Comment (pmarsh-scottlogic) [inline frontend/src/components/ThemedInput/ThemedNumberInput.css:8]: super duper nitpicky but can this comment go next to the `appearance: textfield;` line

File: frontend/src/components/ControlPanel/ControlPanel.tsx [hunk 2]
```diff
@@ -59,53 +60,52 @@ function ControlPanel({
 		MODEL_DEFENCES.includes(defence.id)
 	);
 
-	// only allow configuration in sandbox
-	const showConfigurations = currentLevel === LEVEL_NAMES.SANDBOX;
+	// only allow configuration in sandbox and level 3
+	const showConfigurations =
+		currentLevel === LEVEL_NAMES.SANDBOX ||
+		currentLevel === LEVEL_NAMES.LEVEL_3;
```
Comment (pmarsh-scottlogic) [inline frontend/src/components/ControlPanel/ControlPanel.tsx:66]: this is a fairly major design change! Before we were allowing defences toggling on and off but no configuration of the config item values.

Also we're still hiding the `<ModelBox>` in level 3. A user might wonder why those are missing from level 3 when all the other defences are available.

though it's hard for me to oppose the change, given the defences have almost zero practical relevance to the level as it is currently set up.

File: frontend/src/components/ThemedButtons/DetailElement.tsx
Comment (pmarsh-scottlogic) [inline frontend/src/components/ThemedButtons/DetailElement.tsx:1]: super nice

File: frontend/src/components/ControlPanel/ControlPanel.css [hunk 3]
```diff
@@ -12,26 +15,32 @@
...
+.control-panel .defence-config-container,
+.control-panel .model-config-container {
 	border-radius: 0.5rem;
-	background-color: var(--control-body-background-colour);
+	background-color: var(--control-header-background-colour);
+}
+
+.control-panel h2 {
+	margin: 0;
+	font-size: 1.25rem;
```
[the remark this answers is not in the document]
Comment (chriswilty) [reply frontend/src/components/ControlPanel/ControlPanel.css:26]: Ah yes, I changed that but forgot to remove these 👍

File: frontend/src/components/ControlPanel/ControlPanel.tsx (hunk 2 above)
Comment (chriswilty) [reply frontend/src/components/ControlPanel/ControlPanel.tsx:66]: Ooh, I think that was a mistake, possibly a merge conflict gone wrong... I'll fix that now.

File: frontend/src/components/ControlPanel/ControlPanel.tsx [hunk 4]
```diff
@@ -59,53 +60,52 @@ function ControlPanel({
...
+								currentLevel={currentLevel}
+								defences={nonModelDefences}
+								showConfigurations={showConfigurations}
+								resetDefenceConfiguration={resetDefenceConfiguration}
+								toggleDefence={toggleDefence}
+								setDefenceConfiguration={setDefenceConfiguration}
+							/>
+						</DetailElement>
+					</div>
+					<div className="model-config-container">
```
[the remark this answers is not in the document]
Comment (chriswilty) [reply frontend/src/components/ControlPanel/ControlPanel.tsx:89]: Hmm....

File: frontend/src/components/ControlPanel/ControlPanel.tsx [hunk 5]
```diff
@@ -59,7 +60,7 @@ function ControlPanel({
 		MODEL_DEFENCES.includes(defence.id)
 	);
 
-	// only allow configuration in sandbox
+	// only allow configuration in sandbox and level 3
```
Comment (pmarsh-scottlogic) [inline frontend/src/components/ControlPanel/ControlPanel.tsx:63]:
```suggestion
	// only allow configuration in sandbox
```

**Review by pmarsh-scottlogic — CHANGES_REQUESTED** [review]
Nice nice, just one more thing

**Pull request merged**